src/main/scala/treadle/executable/ExecutionEngine.scala

```diff
@@ -478,21 +480,33 @@ class ExecutionEngine(
     }
   }
 
-  /**
-    * Is the circuit currently stopped.  StopOp throws a
+  /** Is the circuit currently stopped.  StopOp throws a
     * Stop
     * @return
     */
   def stopped: Boolean = {
     lastStopResult.isDefined
   }
 
+  /*
+  This is where things should go that need to be done at the end of the run.
+  Currently this telling black boxes to finish up things if they need to
+  and have the coverage counts recorded if they exist.
+   */
   def finish(): Unit = {
     symbols.foreach { symbol =>
       symbolTable.getBlackboxImplementation(symbol).foreach { blackBox =>
         blackBox.finish()
       }
     }
+    if (symbolTable.verifyOps.nonEmpty) {
+      val text = symbolTable.verifyOps.map { verifyOp =>
+        s"${verifyOp.info.toString.trim},${verifyOp.message.escape},${verifyOp.clockCount},${verifyOp.coverCount}"
+      }.sorted.mkString("\n")
+      val writer = new PrintWriter(new File(generatedFileRoot + ".coverage.txt"))
+      writer.write(text)
+      writer.close()
```

Comment:
Instead of writing this to a file, could you just collect the coverage into an annotation?

Something like:
`case class CoverageResultAnnotation(counts: Map[String, Int]) extends NoTargetAnnotation`

Reply:
Sorry, I meant `Map[String, Long]`, not `Int`.

Reply:
Sure, that would be easy. In general though simulators do not return annotations. I think we need to discuss how this should be implemented

Reply:
I looked at the `TreadleBackend` in `chiseltest` and I think it should be enough to add a `getCoverage: Map[String,Long]` method to the `TreadleTester` class.